### PR TITLE
Refactor text conditions

### PR DESCRIPTION
**Pull request by fokinp**
## Proposed changes

1. Move common text conditions logic to base class
   
- save/cache actual value/text that is used for compassing and for debug purpose in case of failure
- simplify TextCondition extension

2. Refactor unit test related to text conditions

## Checklist
- [ ] Checkstyle and unit tests are passed locally with my changes by running `gradlew check chrome_headless firefox_headless` command
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by asolntsev** [comment]
@fokinp Please fix the tests

**Review by asolntsev — CHANGES_REQUESTED** [review]
It's a great initiative to refactor the duplicated code!

But there is a problem with storing actual value in `Condition`: it's not thread-safe. People can store a single `Condition` instance, say, to a static variable, and use it in multiple tests, and even in multiple parallel threads.

**Comment by fokinp** [comment]
> But there is a problem with storing actual value in `Condition`: it's not thread-safe. People can store a single `Condition` instance, say, to a static variable, and use it in multiple tests, and even in multiple parallel threads.

@asolntsev that is very strange way to use conditions, but I got the point anyway :).
In this case I see two ways:

1.  do not cache actual value and just call `getText` second time for `actualValue`
2.  or use `ThreadLocal` storage for actual value per thread

If we go with 1st option:
 * potentially can be a case when value/text be changed after `apply` call (e.g. https://github.com/selenide/selenide/issues/217) and this behavior can confuse someone who will read/analyze test run
 * we unnecessary do an additional browser interaction (very minor issue)
 * we need do refactoring of: `SelectedText`

And small additional point: will be nice do not check comparing logic in condition tests (like we currently do) and cover this in `HtmlTest` and check only that we call correct method from `Html` class and pass correct values to it. What do you think?

**Comment by asolntsev** [comment]
@fokinp Yes, #1 is not good (at least because of https://github.com/selenide/selenide/issues/217). Currently I don't see any other solutions except #2 (`ThreadLocal `). Let's do it.

File: src/main/java/com/codeborne/selenide/conditions/TextCondition.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.codeborne.selenide.Driver;
+import org.openqa.selenium.WebElement;
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+@ParametersAreNonnullByDefault
+public abstract class TextCondition extends Condition {
+  private final String expectedText;
+  private final ThreadLocal<String> actualText = new ThreadLocal<>();
```
Comment (fokinp) [inline src/main/java/com/codeborne/selenide/conditions/TextCondition.java:13]: @asolntsev jfyi, I get `"ThreadLocal" variables should be cleaned up when no longer used` sonar rule violation locally. I don't know exactly it is important for us or no

**Comment by asolntsev** [comment]
because of merge conflicts, the work continues in a new PR https://github.com/selenide/selenide/pull/1566

@fokinp Thank you for the PR and sorry for so long delay. :(

**Pull request closed without merging**